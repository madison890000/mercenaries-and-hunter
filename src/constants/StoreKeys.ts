export const RESUME_SUMMARY = 'resume-summary';
export const SCORE_KEY = 'score';
export const ADVISE_KEY = 'advise';
export const I18N_KEY = 'i18n';

export const AUTO_CL_COMPANY_NAME = 'auto-cl-company-name';
export const AUTO_CL_JOB_ADDRESS = 'auto-cl-job-address';
export const AUTO_CL_AI_MESSAGES = 'auto-cl-ai-messages';
export const TRANSLATE_ORIGIN_DATA = 'translate-origin-data';
export const TRANSLATE_AI_TRANSLATED_DATA = 'translate-ai-translated-data';
export const GOOGLE_TOKEN_KEY = 'google-token';
