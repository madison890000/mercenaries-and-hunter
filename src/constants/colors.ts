const COLORS = [
    'var(--color-red)',
    'var(--color-yellow)',
    'var(--color-dark-yellow)',
    'var(--color-dark-red)',
    'var(--color-green)',
    '#FF66B2'
] as const;

export const BACKGROUND_OF_MOUNT_RECT = '#E0E0E0';

export default COLORS;
