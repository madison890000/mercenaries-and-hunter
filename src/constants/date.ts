export const ONE_WEEK = 60 * 1000 * 60 * 24 * 7;
