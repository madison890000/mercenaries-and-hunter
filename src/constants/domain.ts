import {IS_DEV} from "./environment";

export const API_DOMAIN =
    IS_DEV ?
        'http://localhost:8083'
        : `https://chat.mercenarieshunter.com`;

export const CHROME_EXTENSION_LINK_ADDRESS = "https://chrome.google.com/webstore/detail/it-mercenaries-and-hunter/eilakanollhbgdoppbffeikcbkhmeloc?hl=zh-CN&authuser=0";

export const LINKEDIN_OFFICIAL_WEB_ADDRESS = "https://www.linkedin.com/"
export const FREELANCER_OFFICIAL_WEB_ADDRESS = "https://www.freelancer.com/"
export const UPWORK_OFFICIAL_WEB_ADDRESS = "https://www.upwork.com/"
export const WEB3_CAREER_OFFICIAL_WEB_ADDRESS = "https://web3.career/"