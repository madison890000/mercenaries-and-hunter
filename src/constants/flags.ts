import {IS_DEV} from "./environment";

const FLAGS = {
}
export default FLAGS