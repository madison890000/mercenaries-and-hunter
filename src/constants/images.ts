import {LocalesKey} from "../i18n/languages";


export const SCREEN_SHOT_IMAGES_WITH_LOCALE = {
    linkedin: {
        'ja-JP': '/imgs/linkedin-ja-JP.png',
        default: "/imgs/linkedin.png",
    },
    upwork: {
        'ja-JP': '/imgs/upwork-ja-JP.png',
        default: "/imgs/upwork.png",
    },
    freelancer: {
        'ja-JP': '/imgs/freelancer-ja-JP.png',
        default: "/imgs/freelancer.png",
    },
    web3Career: {
        default: "/imgs/web3.png",
    }
} as unknown as {
    linkedin: Record<LocalesKey | 'default', string>
    upwork: Record<LocalesKey | 'default', string>
    freelancer: Record<LocalesKey | 'default', string>
    web3Career: Record<LocalesKey | 'default', string>
};

export const HELP_IMAGES = {
    defaultExtensionHome: '/imgs/help-1.png',
    defaultExtensionPopup: '/imgs/help-2.png',
    markSiteInExtensionPopup: '/imgs/help-3.png',
    appliedListButtonInExtensionPopup: '/imgs/help-4.png',
    appliedListInWeb: '/imgs/help-5.png',
}