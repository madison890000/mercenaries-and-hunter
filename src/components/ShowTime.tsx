import {defineMessages, useIntl} from "react-intl";
import {ONE_DAY, ONE_HOUR, ONE_MIN, ONE_MONTH, SEVEN_DAY, TWO_WEEK_DAY} from "../constants/date";


const messages = defineMessages({
    now: {
        id: 'time.now',
    },
    min5: {
        id: 'time.5mins',
    },
    min30: {
        id: 'time.half-hour',
    },
    min60: {
        id: 'time.one-hour',
    },
    hour2: {
        id: 'time.two-hour',
    },
    hour6: {
        id: 'time.six-hour',
    },
    hour12: {
        id: 'time.half-day',
    },
    hour24: {
        id: 'time.one-day',
    },
    day7: {
        id: 'time.one-week',
    },
    day14: {
        id: 'time.two-week',
    },
    day31: {
        id: 'time.one-mouth',
    },
});
const TimeUntilNow: React.FC<{ time: string }> = ({time}) => {
    const now = Date.now();
    const timeSeconds = now - new Date(time).getTime();
    const intl = useIntl();
    let text = '-';
    if (timeSeconds < ONE_MIN) {
        text = intl.formatMessage(messages.now)
    }else if (timeSeconds < ONE_MIN * 5) {
        text = intl.formatMessage(messages.min5)
    }else if (timeSeconds < ONE_MIN * 30) {
        text = intl.formatMessage(messages.min30)
    }else if (timeSeconds < ONE_HOUR) {
        text = intl.formatMessage(messages.min60)
    }else if (timeSeconds < ONE_HOUR * 2) {
        text = intl.formatMessage(messages.hour2)
    }else if (timeSeconds < ONE_HOUR * 6) {
        text = intl.formatMessage(messages.hour6)
    }else if (timeSeconds < ONE_HOUR * 12) {
        text = intl.formatMessage(messages.hour12)
    }else if (timeSeconds < ONE_DAY) {
        text = intl.formatMessage(messages.hour24)
    }else if (timeSeconds < SEVEN_DAY) {
        text = intl.formatMessage(messages.day7)
    }else if (timeSeconds < TWO_WEEK_DAY) {
        text = intl.formatMessage(messages.day14)
    }else if (timeSeconds < ONE_MONTH) {
        text = intl.formatMessage(messages.day31)
    }
    return <>{text}</>
}
const ShowTimeUntilNow: React.FC<{ time: string }> = ({time}) => <span
    style={{
        color: 'gray',
        fontSize: '12px',
        fontStyle: 'italic'
    }}><TimeUntilNow time={time}/></span>
export default ShowTimeUntilNow
