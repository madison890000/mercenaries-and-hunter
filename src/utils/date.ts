import {MONTH_NUMBER_OF_ONE_YEAR, READABLE_MONTHS} from '../constants/date';
import {Dayjs} from "dayjs";

const translateToReadableMonth = (month: number) => READABLE_MONTHS[month]?.slice(0, 3);

export const formatDate = (time: Dayjs) => {
    return `${translateToReadableMonth(time.month())}, ${time.year()}`;
};

const getFullMonth = (time: Dayjs) => {
    return time.year() * MONTH_NUMBER_OF_ONE_YEAR + time.month();
};

export function findPeriodByDate<T>(time: Dayjs, periods: (T & { start: Dayjs; end?: Dayjs })[]) {
    return periods?.find(p => {
        const notBefore = getFullMonth(p?.start) <= getFullMonth(time);
        const notAfter = getFullMonth(p?.end ?? new Dayjs()) >= getFullMonth(time);
        return notBefore && notAfter;
    });
}

export const getMonthCountFromStartAndEnd = (start: Dayjs, end: Dayjs) => {
    const years = end.year() - start.year();
    if (years) {
        return MONTH_NUMBER_OF_ONE_YEAR - start.month() + (years - 1) * MONTH_NUMBER_OF_ONE_YEAR + end.month();
    }
    return end.month() - start.month();
};
