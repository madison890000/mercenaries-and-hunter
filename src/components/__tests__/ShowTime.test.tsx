import React from 'react';
import ShowTime from '../ShowTime';
import renderer from 'react-test-renderer';
import {ONE_DAY, ONE_HOUR, ONE_MIN, ONE_MONTH, SEVEN_DAY} from "../../constants/date";

const FAKE_NOW = 1696039796299
beforeEach(() => {
    jest.useFakeTimers();
    jest.setSystemTime(1696039796299)
});

afterEach(() => {
    jest.useRealTimers()
});

test('ShowTime just now', () => {
    const DividerDom = renderer.create(
        <ShowTime time={new Date(FAKE_NOW - 20).toISOString()}/>
    ).toJSON();
    expect(DividerDom).toMatchSnapshot();
});

test('ShowTime within 5 mins', () => {
    const DividerDom = renderer.create(
        <ShowTime time={new Date(FAKE_NOW - ONE_MIN * 4).toISOString()}/>
    ).toJSON();
    expect(DividerDom).toMatchSnapshot();
});

test('ShowTime within 30 mins', () => {
    const DividerDom = renderer.create(
        <ShowTime time={new Date(FAKE_NOW - ONE_MIN * 29).toISOString()}/>
    ).toJSON();
    expect(DividerDom).toMatchSnapshot();
});

test('ShowTime within 1 hours', () => {
    const DividerDom = renderer.create(
        <ShowTime time={new Date(FAKE_NOW - ONE_MIN * 59).toISOString()}/>
    ).toJSON();
    expect(DividerDom).toMatchSnapshot();
});

test('ShowTime within 2 hours', () => {
    const DividerDom = renderer.create(
        <ShowTime time={new Date(FAKE_NOW - ONE_HOUR * 2 + ONE_MIN).toISOString()}/>
    ).toJSON();
    expect(DividerDom).toMatchSnapshot();
});

test('ShowTime within 6 hours', () => {
    const DividerDom = renderer.create(
        <ShowTime time={new Date(FAKE_NOW - ONE_HOUR * 6 + ONE_MIN).toISOString()}/>
    ).toJSON();
    expect(DividerDom).toMatchSnapshot();
});


test('ShowTime within 12 hours', () => {
    const DividerDom = renderer.create(
        <ShowTime time={new Date(FAKE_NOW - ONE_HOUR * 12 + ONE_MIN).toISOString()}/>
    ).toJSON();
    expect(DividerDom).toMatchSnapshot();
});

test('ShowTime within 1 day', () => {
    const DividerDom = renderer.create(
        <ShowTime time={new Date(FAKE_NOW - ONE_DAY + ONE_MIN).toISOString()}/>
    ).toJSON();
    expect(DividerDom).toMatchSnapshot();
});

test('ShowTime within 7 day', () => {
    const DividerDom = renderer.create(
        <ShowTime time={new Date(FAKE_NOW - SEVEN_DAY + ONE_MIN).toISOString()}/>
    ).toJSON();
    expect(DividerDom).toMatchSnapshot();
});

test('ShowTime within 30 day', () => {
    const DividerDom = renderer.create(
        <ShowTime time={new Date(FAKE_NOW - ONE_MONTH + ONE_MIN).toISOString()}/>
    ).toJSON();
    expect(DividerDom).toMatchSnapshot();
});




