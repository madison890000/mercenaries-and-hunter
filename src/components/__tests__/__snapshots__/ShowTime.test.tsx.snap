// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowTime just now 1`] = `
<span
  style={
    Object {
      "color": "gray",
      "fontSize": "12px",
      "fontStyle": "italic",
    }
  }
>
  Now
</span>
`;

exports[`ShowTime within 1 day 1`] = `
<span
  style={
    Object {
      "color": "gray",
      "fontSize": "12px",
      "fontStyle": "italic",
    }
  }
>
  today
</span>
`;

exports[`ShowTime within 1 hours 1`] = `
<span
  style={
    Object {
      "color": "gray",
      "fontSize": "12px",
      "fontStyle": "italic",
    }
  }
>
  one hour ago
</span>
`;

exports[`ShowTime within 2 hours 1`] = `
<span
  style={
    Object {
      "color": "gray",
      "fontSize": "12px",
      "fontStyle": "italic",
    }
  }
>
  two hour ago
</span>
`;

exports[`ShowTime within 5 mins 1`] = `
<span
  style={
    Object {
      "color": "gray",
      "fontSize": "12px",
      "fontStyle": "italic",
    }
  }
>
  5 minutes ago
</span>
`;

exports[`ShowTime within 6 hours 1`] = `
<span
  style={
    Object {
      "color": "gray",
      "fontSize": "12px",
      "fontStyle": "italic",
    }
  }
>
  6 hour ago
</span>
`;

exports[`ShowTime within 7 day 1`] = `
<span
  style={
    Object {
      "color": "gray",
      "fontSize": "12px",
      "fontStyle": "italic",
    }
  }
>
  this week
</span>
`;

exports[`ShowTime within 12 hours 1`] = `
<span
  style={
    Object {
      "color": "gray",
      "fontSize": "12px",
      "fontStyle": "italic",
    }
  }
>
  12 hour ago
</span>
`;

exports[`ShowTime within 30 day 1`] = `
<span
  style={
    Object {
      "color": "gray",
      "fontSize": "12px",
      "fontStyle": "italic",
    }
  }
>
  this month
</span>
`;

exports[`ShowTime within 30 mins 1`] = `
<span
  style={
    Object {
      "color": "gray",
      "fontSize": "12px",
      "fontStyle": "italic",
    }
  }
>
  half an hour ago
</span>
`;
